mlir/test/Quantum/AdjointTest.mlir

```diff
@@ -109,7 +109,7 @@ func.func @workflow_unhandled() {
   ^bb0(%arg0: !quantum.reg):
     %qb = quantum.extract %arg0[0] : !quantum.reg -> !quantum.bit
     // expected-error@+1 {{Unhandled operation in adjoint region}}
-    quantum.measure %qb : i1, !quantum.bit
+    quantum.measure %qb {postselect = -1 : i8} : i1, !quantum.bit
```

Comment:
I know there were some previous comments regarding this, so maybe you could summarize what the arguments were for always including this attribute.
I would argue that this is against the "style" for the MLIR representation to include a "null value" for attribute instead of outright omitting it, since there is native support for optional attributes. The Python bindings for an operation with optional arguments should also allow you to easily construct the operation with and without the attribute present.

Reply:
I think it was because we always send a value from the frontend to the MLIR, so, having an optional attribute wouldn't be necessary:
```
let arguments = (ins
        QubitType:$in_qubit,
        I8Attr:$postselect
    );
```
However, I agree with your concerns and would prefer to have it as optional:
```
let arguments = (ins
        QubitType:$in_qubit,
         OptionalAttr<I8Attr>:$postselect
    );
```
All I have to do is drop this commit: https://github.com/PennyLaneAI/catalyst/pull/491/commits/08b81623e8f472e8bfe6210a01370c32ffc61683

@rmoyard what do you think?

Reply:
> I think it was because we always send a value from the frontend to the MLIR

Can the attribute not remain `None` in the frontend when the user doesn't specify it? That would then translate to also omitting it in the MLIR representation.

Reply:
Done. I have made the postselect attribute optional again.

Reply:
From your current PR description it sounds like -1 is an allowed value at MLIR level. Isn't that redundant since the attribute is now optional?

Reply:
I have updated the PR description. I tried to make two different bindings at the frontend, one with **postselect** and the previous one without it. The problem was that the generated LLVM IR function calls for the latter did not include the **postselect** attribute, which is mandatory for the **measure** function at the runtime, hence raising an error. Any idea on how to fix that?

Reply:
Don't these lines already take care of generating the -1 value when the attribute is absent?
```cpp
        // Create the postselect value. If not given, it defaults to -1
        LLVM::ConstantOp postselect = rewriter.create<LLVM::ConstantOp>(
            loc, op.getPostselect() ? op.getPostselectAttr() : rewriter.getI8IntegerAttr(-1));
```

Reply:
Yes. But at the frontend I do the same:
```
# Check if the postselect value was given, otherwise default to -1
postselect_attr = op.in_classical_tracers[1] if len(op.in_classical_tracers) == 2 else -1

qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=postselect_attr)
```
because the following:

`qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit)`

generates a function call to measure without postselect, which is mandatory

Reply:
Thanks for the suggestions. I think this commit solves the redundancy issue https://github.com/PennyLaneAI/catalyst/commit/466489e443ebe6bdba570d31d9435ab2f440c0af